llvm/lib/Transforms/Utils/LoopPeel.cpp

```diff
@@ -217,33 +218,39 @@ PhiAnalyzer::PeelCounter PhiAnalyzer::calculate(const Value &V) {
   if (L.isLoopInvariant(&V))
     // Loop invariant so known at start.
     return (IterationsToInvariance[&V] = 0);
-  if (const PHINode *Phi = dyn_cast<PHINode>(&V)) {
+  if (PHINode *Phi = dyn_cast<PHINode>(&V)) {
     if (Phi->getParent() != L.getHeader()) {
       // Phi is not in header block so Unknown.
       assert(IterationsToInvariance[&V] == Unknown && "unexpected value saved");
       return Unknown;
     }
+
+    // If Induction PHI, register as a starting point.
```

Comment:
I think a comment explaining some more context would be useful. May be copy the comments from the test here too as they provide the motivation?

Reply:
@madhur13490  Thanks for your comment. 
Comment added. How about something like this?

Reply:
Thanks. Looks good.

Have you measured the impact of the patch on key benchmarks? LLVM test-suite is a suite to make sure we don't regress.